WcaOnRails/app/webpacker/components/CompetitionsOverview/CompetitionsView.js

```diff
@@ -11,23 +11,42 @@ import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken'
 import CompetitionsFilters from './CompetitionsFilters';
 import ListView from './ListView';
 import MapView from './MapView';
-import { filterReducer, filterInitialState } from './filterUtils';
+import {
+  createFilterState,
+  filterReducer,
+  getDisplayMode,
+  updateSearchParams,
+} from './filterUtils';
 import { calculateQueryKey, createSearchParams } from './queryUtils';
 import useDebounce from '../../lib/hooks/useDebounce';
 import { isCancelled, isInProgress, isProbablyOver } from '../../lib/utils/competition-table';
 
 const DEBOUNCE_MS = 600;
 
 function CompetitionsView() {
-  const [filterState, dispatchFilter] = useReducer(filterReducer, filterInitialState);
+  const searchParams = useMemo(
+    () => new URLSearchParams(window.location.search),
+    [],
+  );
+
+  const [filterState, dispatchFilter] = useReducer(
+    filterReducer,
+    searchParams,
+    createFilterState,
+  );
   const debouncedFilterState = useDebounce(filterState, DEBOUNCE_MS);
-  const [displayMode, setDisplayMode] = useState('list');
+  const [displayMode, setDisplayMode] = useState(() => getDisplayMode(searchParams));
```

Comment:
Why are you passing this as a callback, instead of `useState(getDisplayMode(searchParams))`?

Reply:
No reason in particular. It saves the function `getDisplayMode` from evaluating, but that's negligible.

Reply:
TIL https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state is a thing, thanks :)